src/model_clay.py

```diff
@@ -892,13 +893,25 @@ def predict_step(
         )
         assert not torch.isnan(embeddings_raw).any()  # ensure no NaNs in embedding
 
-        # Take the mean of the embeddings along the sequence_length dimension
-        # excluding the last two latlon_ and time_ embeddings, i.e. compute
-        # mean over patch embeddings only
-        embeddings_mean: torch.Tensor = embeddings_raw[:, :-2, :].mean(dim=1)
-        assert embeddings_mean.shape == torch.Size(
-            [self.model.encoder.B, 768]  # (batch_size, hidden_size)
-        )
+        if self.hparams.output_patch_embeddings:
+            embeddings_raw = rearrange(
+                embeddings_raw[:, :-2, :], "b (w h g) s -> b (w h s) g", w=16, h=16, g=6
+            )
+            embeddings_mean: torch.Tensor = embeddings_raw.mean(dim=2)
```

Comment:
We can use the `reduce` operation from einops, that is more explicit.

For eg: `reduce(embeddings_raw, "b g h w d -> b h w d", "mean")`

Reply:
good idea, Einstein is the solution for everything 🧑‍🔬  😸